### PR TITLE
fix(axe-core): resolved accessibility errors

### PR DESCRIPTION
**Pull request by IgnacioBecerra**
### Related Ticket(s)
#8351
#8353

### Description
Resolved the bugs that the Axe Core checker flagged upon running the tests in both React and WC.

### Changelog

**New**

- {{new thing}}

**Changed**

- changed/removed some `aria` or `role` attributes in some components to ensure the tests passed

### CODE REVIEW

**Review by kennylam — APPROVED** [review]
Tests passing now, thanks @IgnacioBecerra!

File: packages/react/src/components/Masthead/MastheadL1.js [hunk 1]
```diff
@@ -29,6 +29,16 @@ const MastheadL1 = ({ navigationL1, ...rest }) => {
   });
   const childLinkChecker = rest.hasCurrentUrl();
 
+  useEffect(() => {
+    document
+      .querySelector('.bx--header__menu-bar')
```
Comment (kennylam) [inline packages/react/src/components/Masthead/MastheadL1.js:34]:
```suggestion
      .querySelector(`.${prefix}--header__menu-bar`)
```

File: packages/react/src/components/Masthead/MastheadL1.js [hunk 2]
```diff
@@ -29,6 +29,16 @@ const MastheadL1 = ({ navigationL1, ...rest }) => {
...
+      .setAttribute('role', 'menu');
+    document.querySelectorAll('.bx--header__menu-bar li').forEach(e => {
```
Comment (kennylam) [inline packages/react/src/components/Masthead/MastheadL1.js:36]:
```suggestion
    document.querySelectorAll(`.${prefix}--header__menu-bar li`).forEach(e => {
```

File: packages/react/src/components/Masthead/MastheadTopNav.js [hunk 3]
```diff
@@ -29,6 +29,16 @@ const MastheadTopNav = ({ navigation, ...topNavProps }) => {
 
   const childLinkChecker = topNavProps.hasCurrentUrl();
 
+  useEffect(() => {
+    document
+      .querySelector('.bx--header__menu-bar')
```
Comment (kennylam) [inline packages/react/src/components/Masthead/MastheadTopNav.js:34]:
```suggestion
      .querySelector(`.${prefix}--header__menu-bar`)
```

File: packages/react/src/components/Masthead/MastheadTopNav.js [hunk 4]
```diff
@@ -29,6 +29,16 @@ const MastheadTopNav = ({ navigation, ...topNavProps }) => {
...
+      .setAttribute('role', 'menu');
+    document.querySelectorAll('.bx--header__menu-bar li').forEach(e => {
```
Comment (kennylam) [inline packages/react/src/components/Masthead/MastheadTopNav.js:36]:
```suggestion
    document.querySelectorAll(`.${prefix}--header__menu-bar li`).forEach(e => {
```

**Review by kennylam — COMMENTED** [review]
Forgot to add these in my last review.

**Comment by kennylam** [comment]
@IgnacioBecerra looks like it's failing the prettier check.

**Comment by ariellalgilmore** [comment]
update snapshots?

**Review by kennylam — APPROVED** [review]
LGTM thanks @IgnacioBecerra!

**Pull request merged**